src/pages/workspace/workflows/approvals/WorkspaceWorkflowsApprovalsEditPage.tsx

```diff
@@ -1,27 +1,100 @@
 import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import ApprovalWorkflowEditor from './ApprovalWorkflowEditor';
 
 type WorkspaceWorkflowsApprovalsEditPageProps = WithPolicyAndFullscreenLoadingProps & StackScreenProps<FullScreenNavigatorParamList, typeof SCREENS.WORKSPACE.WORKFLOWS_APPROVALS_EDIT>;
 
 function WorkspaceWorkflowsApprovalsEditPage({policy, isLoadingReportData = true, route}: WorkspaceWorkflowsApprovalsEditPageProps) {
+    const styles = useThemeStyles();
     const {translate} = useLocalize();
+    const [personalDetails] = useOnyx(ONYXKEYS.PERSONAL_DETAILS_LIST);
+    const [approvalWorkflow] = useOnyx(ONYXKEYS.APPROVAL_WORKFLOW);
+    const [initialApprovalWorkflow, setInitialApprovalWorkflow] = useState<ApprovalWorkflow | undefined>();
+    const [isDeleteModalVisible, setIsDeleteModalVisible] = useState(false);
+    const formRef = useRef<ScrollView>(null);
 
     // eslint-disable-next-line rulesdir/no-negated-variables
     const shouldShowNotFoundView = (isEmptyObject(policy) && !isLoadingReportData) || !PolicyUtils.isPolicyAdmin(policy) || PolicyUtils.isPendingDeletePolicy(policy);
 
+    const updateApprovalWorkflow = useCallback(() => {
+        if (!approvalWorkflow || !initialApprovalWorkflow) {
+            return;
+        }
+
+        if (!Workflow.validateApprovalWorkflow(approvalWorkflow)) {
+            return;
+        }
+
+        const membersToRemove = initialApprovalWorkflow.members.filter((initialMember) => !approvalWorkflow.members.some((member) => member.email === initialMember.email));
+        Workflow.updateApprovalWorkflow(route.params.policyID, approvalWorkflow, membersToRemove);
+        Navigation.goBack();
+    }, [approvalWorkflow, initialApprovalWorkflow, route.params.policyID]);
+
+    const removeApprovalWorkflow = useCallback(() => {
+        if (!initialApprovalWorkflow) {
+            return;
+        }
+
+        // Remove the approval workflow using the initial data as it could be already edited
+        Workflow.removeApprovalWorkflow(route.params.policyID, initialApprovalWorkflow);
+        Navigation.goBack();
+    }, [initialApprovalWorkflow, route.params.policyID]);
+
+    // Set the initial approval workflow when the page is loaded
+    useEffect(() => {
+        if (!!initialApprovalWorkflow || !policy || !personalDetails) {
+            return;
+        }
+
+        const defaultApprover = policy?.approver ?? policy.owner;
+        const workflows = convertPolicyEmployeesToApprovalWorkflows({
+            employees: policy.employeeList ?? {},
+            defaultApprover,
+            personalDetails,
+        });
+        const currentApprovalWorkflow = workflows.find((workflow) => workflow.approvers.at(0)?.email === route.params.firstApproverEmail);
+
+        if (!currentApprovalWorkflow) {
+            Workflow.clearApprovalWorkflow();
+            return Navigation.goBack(ROUTES.WORKSPACE_WORKFLOWS.getRoute(route.params.policyID));
```

Comment:
We can see this case if trying to edit the approver email in the URL. I think we should display notFoundPage in this case

Reply:
Fixed 👌